libs/vkd3d/state.c

```diff
@@ -4729,6 +4728,78 @@ static bool d3d12_pipeline_state_put_pipeline_to_cache(struct d3d12_pipeline_sta
     return compiled_pipeline;
 }
 
+static VkResult d3d12_pipeline_state_link_pipeline_variant(struct d3d12_pipeline_state *state,
+        const struct vkd3d_pipeline_key *key, const struct vkd3d_format *dsv_format, VkPipelineCache vk_cache,
+        uint32_t dynamic_state_flags, VkPipeline *vk_pipeline)
+{
+    const struct vkd3d_vk_device_procs *vk_procs = &state->device->vk_procs;
+    struct d3d12_graphics_pipeline_state *graphics = &state->graphics;
+    struct vkd3d_fragment_output_pipeline_desc fragment_output_desc;
+    struct vkd3d_vertex_input_pipeline_desc vertex_input_desc;
+    VkPipelineLibraryCreateInfoKHR library_info;
+    VkGraphicsPipelineCreateInfo create_info;
+    VkPipeline vk_libraries[3];
+    uint32_t library_count = 0;
+    unsigned int i;
+    VkResult vr;
+
+    vk_libraries[library_count++] = graphics->library;
```

Comment:
If graphics->library was created with identifiers, this link must also use VK_PIPELINE_CREATE_FAIL_ON_COMPILE_REQUIRED_BIT. If we fail to link, we can convert to SPIR-V and try again.

Reply:
nvm, it's added below.

Reply:
what exactly is this referring to? We are setting `VK_PIPELINE_CREATE_FAIL_ON_COMPILE_REQUIRED_BIT` for linking.